### PR TITLE
NON-OS SDK support. Tested on v2.0.0.

### PR DESCRIPTION
**Pull request by valkuc**
Related discussion - https://github.com/pasko-zh/brzo_i2c/issues/9

### CODE REVIEW

**Comment by pasko-zh** [comment]
Thanks! Will check on the weekend!

File: brzo_i2c.c [hunk 1]
```diff
@@ -708,13 +726,14 @@ void ICACHE_RAM_ATTR brzo_i2c_read(uint8_t *data, uint8_t nr_of_bytes, boolean r
...
 }
 
-void ICACHE_RAM_ATTR brzo_i2c_start_transaction(uint8_t slave_address, uint16_t SCL_frequency_KHz) {
+void ICACHE_FLASH_ATTR brzo_i2c_start_transaction(uint8_t slave_address, uint16_t SCL_frequency_KHz)
+{
 	// 7 Bit Slave Address; SCL Frequency in Steps of 100 KHz, range: 100 -- 1000 KHz
 
 	i2c_slave_address = slave_address;
 	if (i2c_SCL_frequency != SCL_frequency_KHz) {
-		uint16_t fr_sel = round(SCL_frequency_KHz / 100.0);
```
Comment (pasko-zh) [inline brzo_i2c.c]: I would like to keep that with the round(.), such that frequencies are rounded up to next 100 level. Guess that's ok for NON-OS as well.

File: brzo_i2c.c [hunk 2]
```diff
@@ -708,13 +726,14 @@ void ICACHE_RAM_ATTR brzo_i2c_read(uint8_t *data, uint8_t nr_of_bytes, boolean r
 	return;
 }
 
-void ICACHE_RAM_ATTR brzo_i2c_start_transaction(uint8_t slave_address, uint16_t SCL_frequency_KHz) {
+void ICACHE_FLASH_ATTR brzo_i2c_start_transaction(uint8_t slave_address, uint16_t SCL_frequency_KHz)
```
Comment (pasko-zh) [inline brzo_i2c.c]: Although wasting a bit of IRAM, I would like to keep the `brzo_i2c_start_transaction` and its counterpart `bzro_i2c_end_transaction` in IRAM, i.e. not `ICACHE_FLASH_ATTR`. As I learned (the hard way), moving from ROM to RAM can take up 3--4 usec, cf. my issue [here](https://github.com/esp8266/Arduino/issues/1536). There are many use cases, where you want to start i2c transactions right away, thus I would keep RAM attribute.

**Comment by valkuc** [comment]
About round(). I prefer to not use floating point calculation in embedded development when it is possible to implement functionality without it. In your case this can be achieved next way:
`uint16_t fr_sel = (SCL_frequency_KHz + 50) / 100;`
One more reason why I will not use round on native SDK is because to use it, we also need to link "libm" to or code. For some reason that was fail for me. I did not investigate why it fails.

**Pull request merged**